src/vs/workbench/services/decorations/browser/decorationsService.ts

```diff
@@ -79,17 +90,36 @@ class DecorationRule {
 		const { color } = data[0];
 		createCSSRule(`.${this.itemColorClassName}`, `color: ${getColor(theme, color)};`, element);
 
-		// badge
-		const letters = data.filter(d => !isFalsyOrWhitespace(d.letter)).map(d => d.letter);
-		if (letters.length) {
-			createCSSRule(`.${this.itemBadgeClassName}::after`, `content: "${letters.join(', ')}"; color: ${getColor(theme, color)};`, element);
+		// icon (only show first)
+		const icon = data.find(d => d.icon)?.icon;
+		if (icon) {
+			this.createIconCSSRule(icon, color, element, theme);
+		} else {
+			// badge
+			const letters = data.filter(d => !isFalsyOrWhitespace(d.letter)).map(d => d.letter);
+			if (letters.length) {
+				createCSSRule(`.${this.itemBadgeClassName}::after`, `content: "${letters.join(', ')}"; color: ${getColor(theme, color)};`, element);
+			}
+
+			// bubble badge
+			// TODO @misolori update bubble badge to use class name instead of unicode
+			createCSSRule(
+				`.${this.bubbleBadgeClassName}::after`,
+				`content: "\uea71"; color: ${getColor(theme, color)}; font-family: codicon; font-size: 14px; padding-right: 14px; opacity: 0.4;`,
+				element
+			);
 		}
+	}
 
-		// bubble badge
-		// TODO @misolori update bubble badge to use class name instead of unicode
+	private createIconCSSRule(icon: ThemeIcon, color: string | undefined, element: HTMLStyleElement, theme: IColorTheme) {
```

Comment:
privates start `_` in joh-land

Reply:
Oh good lol, guess I must have checked in labels to set what mode I was in.